torch/nn/_functions/thnn/unfold.py

```diff
@@ -0,0 +1,49 @@
+from torch.autograd.function import Function, once_differentiable
+from torch._thnn import type2backend
+
+from . import _all_functions
+
+class Im2Col(Function):
+
+    @staticmethod
+    def forward(ctx, input, kernel_size, dilation, padding, stride):
+
+        ctx.kernel_size = kernel_size
+        ctx.dilation = dilation
+        ctx.padding = padding
+        ctx.stride = stride
+
+        ctx.save_for_backward(input)
+        input = input.contiguous()
+
+        ctx._backend = type2backend[type(input)]
+
+        output = input.new()
+
+        ctx._backend.Im2Col_updateOutput(ctx._backend.library_state,
+                                    input, output,
+                                    kernel_size[0], kernel_size[1],
+                                    dilation[0], dilation[1],
```

Comment:
These (kernel_size, dilation, padding) could be integers instead of tuples, right?

Reply:
No, the tuple values do not need to be equal, i.e. you can dilate by different amounts in different dimensions.